// grunt/Gruntfile.js
module.exports = function(grunt) {

  // Project configuration.
  grunt.initConfig({
    pkg: grunt.file.readJSON('package.json'),
    watch: {
      all: {
        // Replace with whatever file you want to trigger the update from
        // Either as a String for a single entry
        // or an Array of String for multiple entries
        // You can use globing patterns like `css/**/*.css`
        // See https://github.com/gruntjs/grunt-contrib-watch#files
        files: ["C:/wamp/www/projet-recup/app/**/*", "C:/wamp/www/projet-recup/src/**/*", "C:/wamp/www/projet-recup/web/js/**/*"],
        options: {
          livereload: true
        }
      },
	  scripts: {
		  files: ['C:/wamp/www/projet-recup/web/dev/css/test.less'],
		  tasks: ['less'],
		  options: {
			  spawn: false,
		  },
	  },
    },
  
    	
    less: {
        dist: {
            options: {
                compress: true,
                yuicompress: true,
                optimization: 2
            },
            files: { 'C:/wamp/www/projet-recup/web/dev/css/test.css' : 'C:/wamp/www/projet-recup/web/dev/css/test.less' }
        }
    },
    
	  concat: {
	    options: {
	      separator: ';',
	    },
	    dist: {
	      src: ['C:/wamp/www/projet-recup/web/dev/js/**'],
	      dest: 'C:/wamp/www/projet-recup/web/js/javascript.js',
	    },
	  },
	  
	  jshint: {
		    all: ['C:/wamp/www/projet-recup/web/js/javascript.js']
		  },
	  
	  uglify: {
		    options: {
		      mangle: false
		    },
		    my_target: {
		      files: {
		        'C:/wamp/www/projet-recup/web/js/javascript.min.js': ['C:/wamp/www/projet-recup/web/js/javascript.js']
		      }
		    }
		  },
    
	  cssmin: {
		  options: {
		    shorthandCompacting: false,
		    roundingPrecision: -1
		  },
		  target: {
		    files: {
		      'C:/wamp/www/projet-recup/web/css/style.css': 'C:/wamp/www/projet-recup/web/css/style.min.css'
		    }
		  }
		}
  });

  // Load the plugin that provides the "uglify" task.
grunt.loadNpmTasks('grunt-contrib-watch');
grunt.loadNpmTasks('grunt-contrib-less');
grunt.loadNpmTasks('grunt-contrib-concat');
grunt.loadNpmTasks('grunt-contrib-jshint');
grunt.loadNpmTasks('grunt-contrib-uglify');
grunt.loadNpmTasks('grunt-contrib-cssmin');

  // Default task(s).
  grunt.registerTask('default', ['watch']);
  grunt.registerTask('javascript', ['concat', 'uglify', 'jshint']);

};
